src/App.js

```diff
@@ -9,20 +14,50 @@ import './App.css';
 import List from './pages/List/List';
 import AddItem from './pages/AddItem/AddItem';
 import NotFound from './pages/NotFound/NotFound';
+import Home from './pages/Home/Home';
 
 // Components
 import NavMenu from './components/NavMenu/NavMenu';
 
+// Functions
+import getToken from './lib/tokens';
+
 function App() {
+  const [userToken, setUserToken] = useState('');
+
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+    if (token) {
+      setUserToken(token);
+    }
+  }, []);
+
+  function saveToken(e) {
+    e.preventDefault();
+    const token = getToken();
+    localStorage.setItem('token', token);
+    setUserToken(token);
+  }
   return (
     <Router>
       <div className="App container">
         <Switch>
           <Route exact path="/">
-            <List />
+            {userToken ? (
+              <Redirect to="/list" />
```

Comment:
I know we discussed possibly making changes here due to the home page flashing for a moment before the list page renders, but after playing around with it a bit I wasn't able to find a quick remedy for that.

One way to lessen the visual impact for the user would be to add a a loading state to the Home component that shows a spinner (or something of that nature) until it has time to check if the token exists and then render or redirect as needed from there, but that is out of scope for this issue and the current AC 😊